File: java-basics/src/main/java/com/myCompany/ks1/Kata1_1.java
package com.myCompany.ks1;

import com.myCompany.Launch;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

import java.io.IOException;
import java.nio.file.Paths;

public class Kata1_1 {
    public static void main(String[] args) throws IOException {
        SparkSession spark = SparkSession.builder().appName("Java-Basics").master("local[*]")
                .getOrCreate();

        // Read File from local filesystem and create an RDD
        String pathToDataFile = Paths.get(Launch.absolut_path_to_directory, "LICENSE").toString();
        JavaRDD<Row> data = spark.read().text(pathToDataFile).javaRDD();

        // display RDD’s content
        data.foreach(l -> System.out.println(l));
        System.out.println("# lines: "+ data.count());

        spark.stop();
    }
}

File: scala-spark/src/main/scala/com/myCompany/ks1/Kata1_1.scala
package com.myCompany.ks1

import com.myCompany.Launch
import org.apache.spark.sql.SparkSession

import java.nio.file.Paths

object Kata1_1 {
  def main(args: Array[String]): Unit = {
    val spark: SparkSession = SparkSession.builder.appName("scala-Basics").master("local[*]")
      .getOrCreate

    // Read File from local filesystem and create an RDD
    val pathToDataFile = Paths.get(Launch.absolut_path_to_directory, "LICENSE").toString
    val data = spark.read.textFile(pathToDataFile).rdd

    // display RDD’s content
    data.foreach(println)
    System.out.println("# lines: " + data.count)

    spark.stop()
  }
}
